docs/reference/indices/migrate-to-data-stream.asciidoc

```diff
@@ -0,0 +1,94 @@
+[role="xpack"]
+[[indices-migrate-to-data-stream]]
+=== Migrate to data stream API
+++++
+<titleabbrev>Migrate to data stream</titleabbrev>
+++++
+
+Migrates an existing alias with its indices to a new
+<<data-streams,data stream>>.
+
+Data streams require a matching <<indices-templates,index template>>.
+See <<set-up-a-data-stream>>.
+
+////
+[source,console]
+----
+POST idx1/_doc/
+{
+    "message" : "testing",
+    "@timestamp" : "2020-01-01"
+}
+
+POST idx2/_doc/
+{
+    "message" : "testing2",
+    "@timestamp" : "2020-01-01"
+}
+
+POST /_aliases
+{
+  "actions": [
+    {
+      "add": {
+        "index": "idx1",
+        "alias": "my-logs",
+        "is_write_index": true
+      }
+    },
+    {
+      "add": {
+        "index": "idx2",
+        "alias": "my-logs"
+      }
+    }
+  ]
+}
+
+PUT /_index_template/template
+{
+  "index_patterns": ["my-logs*"],
+  "data_stream": { }
+}
+----
+////
+
+[source,console]
+----
+POST /_data_stream/_migrate/my-logs
+----
+// TEST[continued]
+
+////
+[source,console]
+-----------------------------------
+DELETE /_data_stream/my-logs
+DELETE /_index_template/template
+-----------------------------------
+// TEST[continued]
+////
+
+[[indices-migrate-to-data-stream-request]]
+==== {api-request-title}
+
+`POST /_data_stream/_migrate/<alias>`
+
+[[indices-migrate-to-data-stream-api-path-params]]
+==== {api-path-parms-title}
+
+`<alias>`::
++
+--
+(Required, string) Name of the alias to be migrated to a data stream. The
+alias's indices will be hidden and converted to backing indices for the
+new data stream. The alias will be removed, and the new data stream will
+be named `<alias>`.
+
+Aliases must meet the following criteria to be migrated to a data stream:
+
+- Must have an index defined as the <<aliases-write-index,write index>>
+- All indices associated with the alias must have a mapping for the
+`@timestamp` field of type `date` or `date_nanos`
+- Must not have any <<filtered,filters>>
+- Must not have any <<aliases-routing,custom routing>>
```

Comment:
I'd reorder this slightly so the required criteria comes first. I also added a sentence about the write index after conversion.
```suggestion
(Required, string)
Name of the index alias to convert to a data stream. The alias must meet the
following criteria:

- The alias has a <<aliases-write-index,write index>>.
- All indices for the alias have a `@timestamp` field mapping of a `date` or `date_nanos` field type.
- The alias has no <<filtered,filters>>.
- The alias does not use <<aliases-routing,custom routing>>.

{es} removes the alias and creates a data stream with the same name. The alias's
indices become hidden backing indices for the stream. The alias's write index
becomes the stream's write index.
```